src/app/(main)/my-voyage/[teamId]/directory/components/DirectoryComponentWrapper.tsx

```diff
@@ -1,128 +1,141 @@
-  if (index === -1) {
-    return arr;
-  }
-  [arr[index], arr[0]] = [arr[0], arr[index]];
-  return arr;
-}
+import { axiosInstance } from "@/utils/axiosInstance";
+
+// interface FetchTeamDirectoryProps {
+//   teamId: number;
+//   user: User | null;
+// }
+
+// export async function fetchTeamDirectory({
+//   teamId,
+//   user,
+// }: FetchTeamDirectoryProps): Promise<AsyncActionResponse<TeamDirectory>> {
+//   const token = getAccessToken();
+
+//   const fetchTeamDirectoryAsync = () =>
+//     GET<TeamDirectory>(
+//       `api/v1/teams/${teamId}`,
+//       token,
+//       "force-cache",
+//       CacheTag.directory,
+//     );
+
+//   const [res, error] = await handleAsync(fetchTeamDirectoryAsync);
+
+//   if (res) {
+//     updateDirectoryWithCurrentTime(res);
+//     const teamMember = res.voyageTeamMembers;
+//     const elementToSort = teamMember.find(
+//       (element) => element.member.discordId === user?.discordId,
+//     );
+//     moveElementToFirst(teamMember, elementToSort);
+//   }
+
+//   return [res, error];
+// }
+
+// function updateDirectoryWithCurrentTime(data: TeamDirectory) {
+//   return data.voyageTeamMembers.forEach((teamMember) => {
+//     const { timezone } = teamMember.member;
+//     const currentTime = getTimezone(timezone);
+//     teamMember.member.currentTime = currentTime;
+//   });
+// }
+
+// function moveElementToFirst<T>(arr: T[], element: T): T[] {
+//   const index = arr.indexOf(element);
+//   if (index === -1) {
+//     return arr;
+//   }
+//   [arr[index], arr[0]] = [arr[0], arr[index]];
+//   return arr;
+// }
 
 interface TeamDirectoryProps {
   params: {
     teamId: string;
   };
 }
 
-export default async function DirectoryComponentWrapper({
+export default function DirectoryComponentWrapper({
   params,
 }: TeamDirectoryProps) {
-  let teamDirectory: TeamDirectory;
+  // let teamDirectory: TeamDirectory;
   const teamId = Number(params.teamId);
-
-  const [user, error] = await getUser();
-
-  const { errorResponse, data } = await getCurrentVoyageData({
-    user,
-    error,
-    teamId,
-    args: { teamId, user },
-    func: fetchTeamDirectory,
-  });
-
-  if (errorResponse) {
-    return (
-      <ErrorComponent
-        errorType={ErrorType.FETCH_VOYAGE_DATA}
-        message={errorResponse}
-      />
-    );
-  }
-
-  if (data) {
-    const [res, error] = data;
-
-    if (error) {
-      return (
-        <ErrorComponent
-          errorType={ErrorType.FETCH_TEAM_DIRECTORY}
-          message={error.message}
-        />
-      );
-    }
-
-    teamDirectory = res!;
-  } else {
-    redirect("/");
-  }
+  const [teamDirectory, setTeamDirectory] = useState<TeamDirectory>();
+
+  // const [user, setUser] = useState();
+
+  // useEffect(() => {
+  //   const fetchUser = async () => {
+  //     const user = await getUser();
+  //     setUser(user);
+  //   };
+
+  //   fetchUser().catch(console.error);
+  // }, []);
+
+  useEffect(() => {
+    const fetchDirectory = async () => {
```

Comment:
Could we do something like thah maybe?

// Function to fetch team directory
const fetchDirectory = async () => {
  try {
    const response = await axiosInstance.get(`/api/v1/teams/${teamId}`);
    setTeamDirectory(response.data);
  } catch (error: any) {
    console.error('Failed to fetch team directory', error);
  }
};

  useEffect(() => {
    fetchDirectory(teamId); // Only call the function
  }, [teamId]);